qiskit/visualization/pass_manager_visualization.py

```diff
@@ -193,3 +193,160 @@ def _get_node_color(pss, style):
+    # appended more than once
+    component_id = 0
+
+    # keep a running count of indexes across stages
+    idx = 0
+
+    prev_node = None
+
+    for st in stages:
+        stage = getattr(pass_manager, st)
+
+        if stage is not None:
+            passes = stage.passes()
+            stagegraph = pydot.Cluster(str(st), label="", fontname="helvetica", labeljust="l")
+            for controller_group in passes:
+
+                # label is the name of the flow controller parameter
+                label = "[{}] {}".format(idx, ", ".join(controller_group["flow_controllers"]))
+
+                # create the subgraph for this controller
+                subgraph = pydot.Cluster(
+                    str(component_id), label=label, fontname="helvetica", labeljust="l"
+                )
+                component_id += 1
+
+                for pass_ in controller_group["passes"]:
+
+                    # label is the name of the pass
+                    node = pydot.Node(
+                        str(component_id),
+                        label=str(type(pass_).__name__),
+                        color=_get_node_color(pass_, style),
+                        shape="rectangle",
+                        fontname="helvetica",
+                    )
+
+                    subgraph.add_node(node)
+                    component_id += 1
+
+                    # the arguments that were provided to the pass when it was created
+                    arg_spec = inspect.getfullargspec(pass_.__init__)
+                    # 0 is the args, 1: to remove the self arg
+                    args = arg_spec[0][1:]
+
+                    num_optional = len(arg_spec[3]) if arg_spec[3] else 0
+
+                    # add in the inputs to the pass
+                    for arg_index, arg in enumerate(args):
+                        nd_style = "solid"
+                        # any optional args are dashed
+                        # the num of optional counts from the end towards the start of the list
+                        if arg_index >= (len(args) - num_optional):
+                            nd_style = "dashed"
+
+                        input_node = pydot.Node(
+                            component_id,
+                            label=arg,
+                            color="black",
+                            shape="ellipse",
+                            fontsize=10,
+                            style=nd_style,
+                            fontname="helvetica",
+                        )
+                        subgraph.add_node(input_node)
+                        component_id += 1
+                        subgraph.add_edge(pydot.Edge(input_node, node))
+
+                    # if there is a previous node, add an edge between them
+                    if prev_node:
+                        subgraph.add_edge(pydot.Edge(prev_node, node))
+
+                    prev_node = node
+
+                stagegraph.add_subgraph(subgraph)
+                idx += 1
```

Comment:
I'm wondering if there is a way to reuse the pass manager drawer for this? There seems to be a lot of duplication with the functionality in `pass_manager_drawer`. It'd be great if we could keep the portion generating the pydot objects the same way and for this function just wrap it in a subgraph.

Reply:
Thinking out loud, there's probably a way to do it where we update the `pass_manager_drawer()` to take an extra boolean parameter, `draw_stages`, to control whether we draw the stages or not. Will start working on the update...

Reply:
I ended up going with a helper function for each subgraph. I also added a helper function for the output to also remove that repetition.